File: subprojects/frida-python/examples/open_service/dtx/processcontrol.py
import sys

import frida


def on_message(message):
    print("on_message:", message)


device = frida.get_usb_device()

control = device.open_service("dtx:com.apple.instruments.server.services.processcontrol")
control.on("message", on_message)
pid = control.request(
    {
        "method": "launchSuspendedProcessWithDevicePath:bundleIdentifier:environment:arguments:options:",
        "args": [
            "",
            "no.oleavr.HelloIOS",
            {},
            [],
            {
                "StartSuspendedKey": False,
            },
        ],
    }
)
control.request({"method": "startObservingPid:", "args": [pid]})

print(f"App spawned, PID: {pid}.  Kill it to see an example message being emitted.")
sys.stdin.read()

File: subprojects/frida-node/examples/open_service/dtx/processcontrol.ts
import frida from "frida";

let control: frida.Service | null = null;

async function main() {
    const device = await frida.getUsbDevice();

    control = await device.openService("dtx:com.apple.instruments.server.services.processcontrol");
    control.message.connect(onMessage);
    const pid = await control.request({
        method: "launchSuspendedProcessWithDevicePath:bundleIdentifier:environment:arguments:options:",
        args: [
            "",
            "no.oleavr.HelloIOS",
            {},
            [],
            {
                StartSuspendedKey: false,
            }
        ]
    });
    await control.request({ method: "startObservingPid:", args: [pid] });

    console.log(`App spawned, PID: ${pid}.  Kill it to see an example message being emitted.`);
}

function onMessage(message: any) {
    console.log("onMessage:", message);
    control!.cancel();
}

main()
    .catch(e => {
        console.error(e);
    });
